api/v1beta1/azurecluster_default.go

```diff
@@ -242,32 +231,32 @@ func (c *AzureCluster) setNodeOutboundLBDefaults() {
 }
 
 func (c *AzureCluster) setControlPlaneOutboundLBDefaults() {
-	// public clusters don't need control plane outbound lb
-	if c.Spec.NetworkSpec.APIServerLB.Type == Public {
-		return
+	lb := c.Spec.NetworkSpec.ControlPlaneOutboundLB
```

Comment:
An unfortunate result of the refactor is that we check for `lb == nil` twice. Since we already ensure in validations that no control plane LB is set when `apiServerLB == public`, I think we can remove that check. We can simplify the code to something like this:
```
// this will be needed in template as well but it's ok to be explicit
if lb == nil {
return
}

lbClass.setControlPlaneDefaults() // set type,sku, idle timeout here

// set frontendip defautlts
if lb.FrontendIPsCount == nil {
}

return
```
wdyt?